openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion/__init__.py

```diff
@@ -166,6 +169,21 @@ def student_view(self, context=None):
         Renders student view for LMS.
         """
         fragment = Fragment()
+        if ENABLE_DISCUSSIONS_MFE.is_enabled(self.course_key) and settings.DISCUSSIONS_MICROFRONTEND_URL:
+            url = f"{settings.DISCUSSIONS_MICROFRONTEND_URL}/discussions/{self.course_key}/topics/{self.discussion_id}"
```

Comment:
Can we move this code to a reusable method to fetch links?
e.g.
```
def get_mfe_discussion_url(course, topic_id=None):
    url = f"{settings.DISCUSSIONS_MICROFRONTEND_URL}/discussions/{self.course_key}/"
    if topic_id:
        url = f'{url}topics/{topic_id}/'
    return url
```

Reply:
I had actually added exactly this but in a different active PR here: https://github.com/edx/edx-platform/pull/29300

If that is merged first, I can update this PR to use that method. A new PR to add support for the in-context UI is also underway, and that will need a URL change here. So we can wait a bit and merge this after that perhaps.